### PR TITLE
[New Resource] : `aws_globalaccelerator_cross_account_attachment`

### PR DESCRIPTION
**Pull request by kathmbeck**
### Description

PR for new resource `aws_globalaccelerator_cross_account_attachment`

### Relations

Closes https://github.com/hashicorp/terraform-provider-aws/issues/34254.

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccGlobalAcceleratorCrossAccountAttachment PKG=globalaccelerator

...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/globalaccelerator/... -v -count 1 -parallel 20 -run='TestAccGlobalAcceleratorCrossAccountAttachment'  -timeout 360m
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_basic
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_basic
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_principals
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_principals
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_resources
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_resources
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_disappears
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_disappears
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_basic
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_resources
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_principals
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_disappears
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_resources (12.14s)
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_disappears (13.06s)
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_basic (15.49s)
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_principals (15.50s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/globalaccelerator  18.282s
```

### CODE REVIEW

**Comment by kathmbeck** [comment]
Hi @ewbankkit! 👋 Wanted to check in here on anything I can do to push this along?

**Comment by coolfrood** [comment]
Hi @ewbankkit! I'm on the AWS Global Accelerator team and I've previously contributed to Global Accelerator's Terraform implementation. In this case, one of our key partners would like to merge in this PR so that they can use a new resource type supported by Global Accelerator. Could you help us?

File: internal/service/globalaccelerator/cross_account_attachment.go [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+	framework.WithTimeouts
+}
+
+func (r *resourceCrossAccountAttachment) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_globalaccelerator_cross_account_attachment"
+}
+
+func (r *resourceCrossAccountAttachment) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
```
Comment (AdamTylerLynch) [inline internal/service/globalaccelerator/cross_account_attachment.go:55]: Attributes should be defined in alphabetical order (not the order that the AWS API defines them in).

File: internal/service/globalaccelerator/cross_account_attachment.go [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+			"attachment_arn": framework.ARNAttributeComputedOnly(),
```
Comment (AdamTylerLynch) [inline internal/service/globalaccelerator/cross_account_attachment.go:56]: We preference using simply `arn` for consistency across resource types.

Cite: https://hashicorp.github.io/terraform-provider-aws/raising-a-pull-request/#resource-contribution-guidelines:~:text=Implements%20arn%20Attribute

"attachment_arn": framework.ARNAttributeComputedOnly(),
```suggestion
			"arn": framework.ARNAttributeComputedOnly(),
```

File: internal/service/globalaccelerator/cross_account_attachment_test.go [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+
+	accountID := ""
+	for i := 0; i < 12; i++ {
+		digit := rand.Intn(10)
+		accountID += strconv.Itoa(digit)
+	}
+	return accountID
+}
+
+func TestExpandResources(t *testing.T) {
```
Comment (AdamTylerLynch) [inline internal/service/globalaccelerator/cross_account_attachment_test.go:39]: @ewbankkit we don't normally see unit test, thought on this?

**Review by AdamTylerLynch — CHANGES_REQUESTED** [review]
What a great first Pull Request! We very much appreciate your contribution! I have requested a few small changes. 

I do appreciate the unit tests, and they provide value, I have asked Hashi engineer @ewbankkit to provide feedback on the unit tests, as we generally only conduct integration tests.

**Comment by kathmbeck** [comment]
@AdamTylerLynch Thank you for the review! I've implemented your suggestions and will await feedback on inclusion of the unit tests.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccEC2EIPDataSource_filter\|TestAccEC2EIP_basic' PKG=ec2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/ec2/... -v -count 1 -parallel 20  -run=TestAccEC2EIPDataSource_filter\|TestAccEC2EIP_basic -timeout 360m
=== RUN   TestAccEC2EIPDataSource_filter
=== PAUSE TestAccEC2EIPDataSource_filter
=== RUN   TestAccEC2EIP_basic
=== PAUSE TestAccEC2EIP_basic
=== CONT  TestAccEC2EIPDataSource_filter
=== CONT  TestAccEC2EIP_basic
--- PASS: TestAccEC2EIPDataSource_filter (18.78s)
--- PASS: TestAccEC2EIP_basic (21.89s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	32.160s
```
```console
% make testacc TESTARGS='-run=TestAccGlobalAcceleratorCrossAccountAttachment_' PKG=globalaccelerator ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.2 test ./internal/service/globalaccelerator/... -v -count 1 -parallel 2  -run=TestAccGlobalAcceleratorCrossAccountAttachment_ -timeout 360m
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_basic
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_basic
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_principals
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_principals
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_resources
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_resources
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_disappears
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_disappears
=== RUN   TestAccGlobalAcceleratorCrossAccountAttachment_tags
=== PAUSE TestAccGlobalAcceleratorCrossAccountAttachment_tags
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_basic
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_disappears
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_disappears (17.46s)
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_tags
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_basic (32.11s)
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_resources
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_tags (43.36s)
=== CONT  TestAccGlobalAcceleratorCrossAccountAttachment_principals
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_principals (28.81s)
--- PASS: TestAccGlobalAcceleratorCrossAccountAttachment_resources (227.81s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/globalaccelerator	269.545s
```

**Comment by ewbankkit** [comment]
@kathmbeck Thanks for the contribution :tada: :clap:.

**Pull request merged**